src/cryptoadvance/specter/addresslist.py

```diff
@@ -48,6 +50,16 @@ def set_label(self, label):
             return
         self["label"] = label
         self.rpc.setlabel(self.address, label)
+    
+    def set_service_id(self, service_id: str):
+        if service_id:
+            # Declares that this Address is associated with a Service
+            self["service_id"] = service_id
+        else:
+            # Frees a reserved Address; reset its label
+            self["service_id"] = None
+            if not self.used and self["label"]:
+                self.set_label("")
```

Comment:
Wouldn't that mean that the address property `is_labeled` is still set to true when using an empty string? An empty string would also not be the "default label" that the address had before (usually sth. like "Address #32").

Reply:
Python bools treat empty string and empty lists as `False`:

```
>>> test = ""
>>> bool(test)
False

>>> if test:
...     print("hello!")
... else:
...     print("nope!")
... 
nope!
```